woocommerce/woocommerce-android
# Issue/6920 payments banner compose UI

Closes: #6920 


### Description

This PR adds a Jetpack Compose UI for the Upsell Card reader banner. Integrates the compose UI with the traditional Android view.


amAw8DJHmOXdF0S40YnAck-fi-191%3A13208





### Testing instructions

1. navigate to any of the orders eligible for IPP.
2. Notice the banner.


#### Upsell card reader banner in the settings screen
1. Make sure your store is set up in either the US or Canada
2. Navigate to the settings screen
3. Notice the Banner
4. Play around with the dismiss and purchase card reader CTA and ensure everything works as expected.

#### Upsell card reader banner in the Payments screen
1. Make sure your store is set up in either the US or Canada
2. Navigate to the payments screen
3. Notice the Banner
4. Play around with the dismiss and purchase card reader CTA and ensure everything works as expected.


#### Upsell card reader banner in the Order list screen
1. Make sure your store is set up in either the US or Canada
2. Navigate to the order list screen
3. Notice the Banner
4. Play around with the dismiss and purchase card reader CTA and ensure everything works as expected.


#### Upsell card reader banner not shown for store set up in any other place apart from the US or Canada
1. Make sure your store is set up in any other country apart from the US or Canada
2. Navigate to the payments screen, Order list screen, Settings screen
3. Notice the Banner is not visible

#### Upsell card reader banner purchase card reader CTA in the US
1. Make sure your store is set up in the US
2. Navigate to the payments screen, Order list screen, Settings screen
3. Notice the Banner
4. Tap on the purchase card reader CTA
5. Ensure M2 card reader details are being opened in the web view


#### Upsell card reader banner purchase card reader CTA in the Canada
1. Make sure your store is set up in the Canada
2. Navigate to the payments screen, Order list screen, Settings screen
3. Notice the Banner
4. Tap on the purchase card reader CTA
5. Ensure WisePad3 card reader details are being opened in the web view


#### Upsell card reader banner purchase card reader "Remind me later" CTA
1. Make sure your store is set up in either the US or Canada
2. Navigate to the payments screen, Order list screen, Settings screen
3. Notice the Banner
4. Tap on the "Remind me later" CTA
5. Ensure the banner appears after the threshold (threshold is 14 days)

For easier testing, modify the code in `SelectPaymentMethodViewModel` lineno: `322` method `isLastDialogDismissedMoreThan14DaysAgo` like this
```
val timeDifference = currentTimeInMillis - getCardReaderUpsellBannerLastDismissed()
        return TimeUnit.MILLISECONDS.toSeconds(timeDifference) > 10
```
The above piece of code basically says the threshold for displaying the banner again after the last dismissal is 5 seconds. After 5 seconds, the banner starts appearing again on the payments screen.


#### Upsell card reader banner purchase card reader "Don't show again" CTA
1. Make sure your store is set up in either the US or Canada
2. Navigate to the payments screen, Order list screen, Settings screen
3. Notice the Banner
4. Tap on the "Don't show again" CTA
5. Ensure the banner never appears on any of these screens until app data is cleared.


### Screenshot

![Screenshot_20220719-151743](https://user-images.githubusercontent.com/1331230/179721690-781a849c-cc03-4d30-92a5-2f4ac2fd8925.png)
